lib/pghero/methods/basic.rb

```diff
@@ -1,48 +1,93 @@
+# frozen_string_literal: true
+
+require 'active_record'
+
 module PgHero
   module Methods
     module Basic
+
+      PG_CONNECTION_ADAPTER_NAMES = %i[postgresql postgis].freeze
+
+      ACTIVE_RECORD_CAST_METHOD = ActiveRecord::VERSION::MAJOR < 5 ? :type_cast : :cast
+
+      private_constant :PG_CONNECTION_ADAPTER_NAMES, :ACTIVE_RECORD_CAST_METHOD
+
+      # from ActiveSupport
+      def self.squish(str)
+        str.to_s.gsub(/\A[[:space:]]+/, '').gsub(/[[:space:]]+\z/, '').gsub(/[[:space:]]+/, ' ')
+      end
+
+      def self.remove_line_comments(sql)
+        sql.gsub(/[\s]*--[^\r\n]*/, '')
+      end
+
+      def self.sql_const(sql)
+        make_squishable(sql, squish(remove_line_comments(sql)))
+      end
+
+      def self.make_squishable(obj, squished, freeze = true)
+        squished = squished.freeze if freeze
+        obj = obj.frozen? ? obj.dup : obj
+        obj.define_singleton_method(:squish, -> { squished })
+        freeze ? obj.freeze : obj
+      end
+
+      def execute(sql)
+        connection.execute(sql)
+      end
+
+      def select_one(sql)
+        select_all(sql).first.values.first
+      end
+
+      def quote(value)
+        connection.quote(value)
+      end
+
       def ssl_used?
         ssl_used = nil
         with_transaction(rollback: true) do
           begin
-            execute("CREATE EXTENSION IF NOT EXISTS sslinfo")
+            execute('CREATE EXTENSION IF NOT EXISTS sslinfo')
           rescue ActiveRecord::StatementInvalid
             # not superuser
           end
-          ssl_used = select_one("SELECT ssl_is_used()")
+          ssl_used = select_one('SELECT ssl_is_used()')
         end
         ssl_used
       end
 
       def database_name
-        select_one("SELECT current_database()")
+        select_one('SELECT current_database()')
       end
 
       def server_version
-        @server_version ||= select_one("SHOW server_version")
+        @server_version ||= select_one('SHOW server_version')
       end
 
       def server_version_num
-        @server_version_num ||= select_one("SHOW server_version_num").to_i
+        @server_version_num ||= select_one('SHOW server_version_num').to_i
       end
```

Comment:
Any functional reason why you're updating all the quote types?

Reply:
Rubocop was complaining about double-quoted strings that didn't require interpolation. It made sense so I updated files I was working on anyway to suppress the warning and make the distinction clear.

Now that you asked it looks like the Ruby style guide leaves this open-ended and the Rubocop setting is configurable in my IDE (RubyMine). I hadn't checked that context and which (if any) model Instacart uses standard.

I can revert the unnecessary changed lines now that I see it's not an agreed Ruby standard:

https://github.com/rubocop-hq/ruby-style-guide#strings

Reply:
String quote style in ruby is a vi/emacs style holy war.